src/ate/test_programs/otlib_wrapper/src/lib.rs

```diff
@@ -72,6 +66,18 @@ pub extern "C" fn OtLibFpgaInit(
     let transport = backend::create(&backend_opts).unwrap();
     transport.apply_default_configuration(None).unwrap();
 
+    Box::into_raw(Box::new(transport))
+}
+
+#[no_mangle]
+pub extern "C" fn OtLibFpgaLoadBitstream(
+    transport: &TransportWrapper,
+    fpga_bitstream: *mut c_char,
+) {
```

Comment:
Technically, the input type is `* const TransportWrapper`, and you should convert it back to a reference in the function:

```
pub extern "C" fn OtLibFpgaLoadBitstream(
    transport: &TransportWrapper,
    fpga_bitstream: *mut c_char,
) {
    let transport = &*transport; // rebind transport as a ref to the pointed-to object.
```

You also might consider marking the whole function as unsafe, as its being called from C (which is an unsafe environment).

```
pub extern "C" unsafe fn Foo(...) { ... }
```

Reply:
fixed.